lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -551,7 +551,9 @@ class MongoClientInterface {
         const { objName, versionId, collectionName } = params;
         if (isUserBucket(collectionName)) {
             if (isUpdate) {
+                const findId = new RegExp(objName);
```

Comment:
Why the need for a regexp?

Reply:
yes but there is a chance it translates to a SCAN as well